src/common/AndroidEnvReqResolver.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { Logger, Messages, SfdxError } from '@salesforce/core';
+import { AndroidUtils } from './AndroidUtils';
+import util from 'util';
+import { CommonUtils } from './CommonUtils';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages(
+    '@salesforce/lwc-dev-mobile-core',
+    'requirement-android'
+);
+
+export type RequirementResolveFunc = () => Promise<string | undefined>;
+export interface RequirementResolver {
+    resolvedMessage: string;
+    unableToResolveMessage: string;
+    logger: Logger;
+    resolveFunction: RequirementResolveFunc;
+}
+
+/**
+ * Resolves the missing Android SDK
+ */
+export class AndroidCommandLineToolsResolver implements RequirementResolver {
+    public resolvedMessage: string;
+    public unableToResolveMessage: string;
+    public logger: Logger;
+    public resolveFunction: RequirementResolveFunc;
+
+    constructor(logger: Logger, resolveFunction?: RequirementResolveFunc) {
+        this.resolvedMessage = messages.getMessage(
+            'android:reqs:cmdlinetools:install:resolvedMessage'
+        );
+
+        this.unableToResolveMessage = messages.getMessage(
+            'android:reqs:cmdlinetools:install:unableToResolveMessage'
+        );
+        this.logger = logger;
+
+        if (typeof resolveFunction !== 'undefined') {
+            this.resolveFunction = resolveFunction;
+        } else {
+            this.resolveFunction = function () {
+                return this.installAndroidCommandLineTools();
+            };
+        }
+    }
+
+    private async installAndroidCommandLineTools(): Promise<
+        string | undefined
+    > {
+        // download command line tools
+
+        // unzip and move to the latest folder
+        const unzipCommand = ''.concat(
+            'unzip /Users/pvandyk/Downloads/commandlinetools-mac-8092744_latest.zip -d $ANDROID_HOME/cmdline-tools/latest',
```

Comment:
`/Users/pvandyk`?

Reply:
Also, not sure I understand what this is doing -- appears it is extracting the zip into $ANDROID_HOME/cmdline-tools/latest, then moving all files from cmdline-tools/latest/cmdline-tools/* into cmdline-tools/latest? 

If so, then perhaps unzip to a tmp folder and just move it to where it belongs?

```
unzip somefile.zip -d $ANDROID_HOME/tmp-cmdline-tools && \
    mkdir -p $ANDROID_HOME/cmdline-tools/latest && \
    mv $ANDROID_HOME/tmp-cmdline-tools $ANDROID_HOME/cmdline-tools/latest
```

Reply:
I maybe able to unzip the folder directly into latest.  I'll look into that, depending on the unzip node solution we choose.